addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -866,93 +867,57 @@ registry['sizing_y'] = registry.sizing.extend({
  */
 registry.colorpicker = SnippetOption.extend({
     xmlDependencies: ['/web_editor/static/src/xml/snippets.xml'],
-    events: _.extend({}, SnippetOption.prototype.events || {}, {
-        'click .colorpicker button': '_onColorButtonClick',
-        'mouseenter .colorpicker button': '_onColorButtonEnter',
-        'mouseleave .colorpicker button': '_onColorButtonLeave',
-        'click .note-color-reset': '_onColorResetButtonClick',
-    }),
-    colorPrefix: 'bg-',
+    custom_events: {
+        color_picked: '_onColorPicked',
+        custom_color_picked: '_onCustomColor',
+        color_hover: '_onColorHovered',
+        color_leave: '_onColorLeft',
+        color_reset: '_onColorReset',
+    },
 
     /**
      * @override
      */
     start: function () {
-        var self = this;
-        var res = this._super.apply(this, arguments);
-
+        const options = {
+            $target: this.$target,
+            $editable: rte.Class.prototype.editable(),
```

Comment:
Clearly, here, all you have to do is trigger_up an event that the editor will receive somewhere and where it can return the editable zone to you without this hack.